xarray/core/computation.py

```diff
@@ -1306,3 +1307,86 @@ def where(cond, x, y):
         dataset_join="exact",
         dask="allowed",
     )
+
+
+def _calc_idxminmax(
+    *,
+    array,
+    func: Callable,
+    dim: Optional[Hashable],
+    skipna: Optional[bool],
+    promote: Optional[bool],
+    keep_attrs: Optional[bool],
+    **kwargs: Any,
+):
+    """Apply common operations for idxmin and idxmax."""
+    # This function doesn't make sense for scalars so don't try
+    if not array.ndim:
+        ValueError("This function does not apply for scalars")
```

Comment:
```suggestion
        raise ValueError("This function does not apply for scalars")
```

It would probably also be a good idea to add a test for these errors just in case :)